app/components/custom_status/custom_status_expiry.tsx

```diff
@@ -0,0 +1,139 @@
+    showPrefix?: boolean;
+    withinBrackets?: boolean;
+}
+
+const CustomStatusExpiry = (props: Props) => {
+    const {time, theme, styleProp, showPrefix, withinBrackets} = props;
+    const timezone = useSelector(getCurrentUserTimezone);
+    const styles = createStyleSheet(theme);
+    const militaryTime = useSelector((state: GlobalState) => getBool(state, Preferences.CATEGORY_DISPLAY_SETTINGS, 'use_military_time'));
+    const currentMomentTime = getCurrentMomentForTimezone(timezone);
+
+    const expiryMomentTime = timezone ? moment(time).tz(timezone) : moment(time);
+
+    const plusSixDaysEndTime = currentMomentTime.clone().add(6, 'days').endOf('day');
+    const tomorrowEndTime = currentMomentTime.clone().add(1, 'day').endOf('day');
+    const todayEndTime = currentMomentTime.clone().endOf('day');
+
+    let useTime = true;
+    let useDay = false;
+    let isTomorrow = false;
+    let isToday = false;
+    let useDate = false;
+    let format = '';
+
+    if (expiryMomentTime.isSame(todayEndTime)) {
+        isToday = true;
+    }
+    if (expiryMomentTime.isAfter(todayEndTime) && expiryMomentTime.isSameOrBefore(tomorrowEndTime)) {
+        isTomorrow = true;
+    }
+    if (expiryMomentTime.isSame(todayEndTime) || expiryMomentTime.isAfter(tomorrowEndTime)) {
+        useTime = false;
+    }
+    if (expiryMomentTime.isBetween(tomorrowEndTime, plusSixDaysEndTime)) {
+        useDay = true;
+    }
+
+    const isCurrentYear = currentMomentTime.get('y') === expiryMomentTime.get('y');
+
+    useDate = !(isToday || useTime || useDay || !isCurrentYear);
+
+    if (useDay) {
+        format = 'dddd';
+    } else if (useDate) {
+        format = 'MMM DD';
+    } else if (!isCurrentYear) {
+        format = 'MMM DD, YYYY';
+    }
+
+    const showDayorDate = (useDay || useDate || !isCurrentYear) ? (
+        <FormattedDate
+            format={format}
+            timezone={timezone}
+            value={expiryMomentTime}
+        />
+    ) : null;
+
+    const showTime = useTime ? (
+        <FormattedTime
+            hour12={!militaryTime}
+            timezone={timezone}
+            value={expiryMomentTime}
+        />
+    ) : null;
+
+    const showTomorrow = isTomorrow ? (
+        <Text>
+            <FormattedText
+                id='custom_status.expiry_time.tomorrow'
+                defaultMessage='Tomorrow'
+            />{' at '}
+        </Text>
+    ) : null;
+
+    const showToday = isToday ? (
+        <FormattedText
+            id='custom_status.expiry_time.today'
+            defaultMessage='Today'
+        />
+    ) : null;
+
+    const prefix = showPrefix ? (
+        <Text>
+            <FormattedText
+                id='custom_status.expiry.until'
+                defaultMessage='Until'
+            />{' '}
+        </Text>
+    ) : null;
+
+    return (
+        <Text
+            testID={props.testID}
+            style={styleProp || styles.text}
+        >
+            {withinBrackets ? '(' : null}
```

Comment:
nit -  I wonder if using `&&` here makes more sense.

```suggestion
            {withinBrackets && '('}
```